src/lightning_app/core/work.py

```diff
@@ -605,3 +605,17 @@ def _register_cloud_compute(self):
         if internal_id not in _CLOUD_COMPUTE_STORE:
             _CLOUD_COMPUTE_STORE[internal_id] = _CloudComputeStore(id=internal_id, component_names=[])
         _CLOUD_COMPUTE_STORE[internal_id].add_component_name(self.name)
+
+    def _apply_flow_delta(self, key: str, new_value: Any, old_value: Any):
+        current_value = getattr(self, key)
+
+        if new_value == current_value:
+            return
+
+        if current_value != old_value:
+            raise Exception(
```

Comment:
Could this get triggered a lot? Maybe safer to just not apply the delta in these cases? That would mimic traditional compare-and-set style stuff

Reply:
Yes, done.